hangupsbot/handlers.py

```diff
@@ -47,23 +47,84 @@ def __init__(self, bot, bot_command='/bot'):
                              {},
                              forgiving=True )
 
-    def register_handler(self, function, type="message", priority=50):
-        """registers extra event handlers"""
+    def register_handler(self, function, type="message", priority=50, extra_metadata={}):
```

Comment:
Agreed, it's a horrible practice to ever use foo=(anything mutable) in a function definition as a default.  Python gets very screwed, and they have a justification for it, but it's just one of those things that will eventually bite you in the ass.

Instead of:
```
def func(foo={}):
    pass
```

I always do:

```
def func(foo=None):
    foo = foo or {}
    pass
```